# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/NodeChannel.cpp
# Commit: c02fa459e77d
# Full Hash: c02fa459e77d2572b2fe740f20d6070a77383589
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-06-22 21:29:07
# Regressor Bug: 1706374
# File Overlap Count: 2
# Description:
#   Bug 1706374 - Part 13: Remove the event footer from an IPC::Message when deserializing, r=handyman,glandium
#   
#   This unfortunately requires a new method to be added to BufferList to
#   support truncating the buffer to a particular iterator.
#   
# ==============================================================================

diff -r 72dc6537cf0b -r c02fa459e77d ipc/glue/NodeChannel.cpp
--- a/ipc/glue/NodeChannel.cpp	Mon Jun 21 21:53:11 2021 +0000
+++ b/ipc/glue/NodeChannel.cpp	Mon Jun 21 21:53:11 2021 +0000
@@ -117,7 +117,7 @@
 
   if (!mClosed) {
     mChannel->Close();
-    MOZ_ALWAYS_TRUE(this == mChannel->set_listener(mExistingListener));
+    mChannel->set_listener(mExistingListener);
   }
   mClosed = true;
 }